gradio/routes.py

```diff
@@ -1243,7 +1243,10 @@ async def component_server(
                     status_code=status.HTTP_404_NOT_FOUND,
                     detail="Function not found.",
                 )
-            return fn(body.data)
+            if inspect.iscoroutinefunction(fn):
```

Comment:
So that you can use async functions in `@server` decorator. Needed for webrtc component.